R/diagnostic_test.R

```diff
@@ -0,0 +1,198 @@
+#' Diagnostic testing
+#'
+#' @description
+#' Run a set of diagnostic tests to check for common issues found when setting
+#' up R on a DfE
+#' system. This includes:
+#'   - Checking for proxy settings in the Git configuration
+#'   - Checking for correct download method used by renv (curl)
+#'
+#' @inheritParams check_proxy_settings
+#'
+#' @return List object of detected parameters
+#' @export
+#'
+#' @examples
+#' diagnostic_test()
+diagnostic_test <- function(
+    clean = FALSE,
+    verbose = FALSE) {
+  results <- c(
+    check_proxy_settings(clean = clean, verbose = verbose),
+    check_github_pat(clean = clean, verbose = verbose),
+    check_renv_download_method(clean = clean, verbose = verbose)
+  )
+  return(results)
+}
+
+#' Check proxy settings
+#'
+#' @description
+#' This script checks for "bad" proxy settings. Prior to the pandemic, analysts
+#' in the DfE would need to add some proxy settings to their GitHub config.
+#' These settings now prevent Git from connecting to remote archives on GitHub
+#' and Azure DevOps if present, so this script identifies and (if clean=TRUE is
+#' set) removes them.
+#'
+#' @param proxy_setting_names Vector of proxy parameters to check for. Default:
+#' c("http.proxy", "https.proxy")
+#' @param clean Attempt to clean settings
+#' @param verbose Run in verbose mode
+#'
+#' @return List of problem proxy settings
+#' @export
+#'
+#' @examples
+#' check_proxy_settings()
+check_proxy_settings <- function(
+    proxy_setting_names = c("http.proxy", "https.proxy"),
+    clean = FALSE,
+    verbose = FALSE) {
+  git_config <- git2r::config()
+  proxy_config <- git_config |>
+    magrittr::extract2("global") |>
+    magrittr::extract(proxy_setting_names)
+  proxy_config <- purrr::keep(proxy_config, !is.na(names(proxy_config)))
+  if (any(!is.na(names(proxy_config)))) {
+    dfeR::toggle_message("Found proxy settings:", verbose = verbose)
+    paste(names(proxy_config), "=", proxy_config, collapse = "\n") |>
+      toggle_message(verbose = verbose)
+    if (clean) {
+      proxy_args <- proxy_config |>
+        lapply(function(list) {
+          NULL
+        })
+      rlang::inject(git2r::config(!!!proxy_args, global = TRUE))
+      message("FIXED: Git proxy settings have been cleared.")
+    } else {
+      message("FAIL: Git proxy setting have been left in place.")
+    }
+  } else {
+    message("PASS: No proxy settings found in your Git configuration.")
```

Comment:
Currently when passing it prints everything whether verbose or not, should it just print the pass message if verbose = FALSE (default)?

![image](https://github.com/user-attachments/assets/5cacb1aa-569b-446f-8ab9-44606212c0f8)

With verbose = FALSE, still shows all information
![image](https://github.com/user-attachments/assets/3509a3a2-c6b3-4882-ad7a-324f0321a6e4)